graal-oap-impl/mal-grammar/src/main/antlr4/org/apache/skywalking/mal/rt/grammar/MALCoreParser.g4

```diff
@@ -0,0 +1,80 @@
+parser grammar MALCoreParser;
+
+options { tokenVocab=MALCoreLexer; }
+
+malExpression: mainExpr;
+
+closure
+ : OPEN_BRACE (CLOSURE_CONTENT|OPEN_CURLY|CLOSE_CURLY)* CLOSE_CURLY
+ ;
+
+
+varDeclare
+ : IDENTIFIER IDENTIFIER (EQUAL groovyExpr)? SEMI?
+ ;
+
+mainExpr:
+        mainExpr DOT methodCall
+        | methodCall
+        | mainExpr op=(STAR | SLASH) mainExpr
+        | mainExpr op=(PLUS | MINUS) mainExpr
+        | IDENTIFIER
+        | NUMBER
+        | OPEN_PAREN mainExpr CLOSE_PAREN
+    ;
+
+groovyExpr:
+        IDENTIFIER DOT IDENTIFIER
+    | CLASS_NAME DOT groovyMethodCall
+    | groovyExpr DOT groovyMethodCall
+    | groovyMethodCall
```

Comment:
Don't put the `groovy**` in new logic. You are going to replace the existing MAL engine, whether it is groovy based doesn't matter.